crates/common/tedge_config/src/tedge_config_cli/tedge_config.rs

```diff
@@ -641,13 +653,20 @@ define_tedge_config! {
     }
 }
 
-fn default_http_address(dto: &TEdgeConfigDto) -> IpAddr {
+fn default_http_bind_address(dto: &TEdgeConfigDto) -> IpAddr {
     let external_address = dto.mqtt.external.bind.address;
     external_address
         .or(dto.mqtt.bind.address)
         .unwrap_or(Ipv4Addr::LOCALHOST.into())
 }
 
+fn default_http_address(dto: &TEdgeConfigDto) -> Arc<str> {
+    let mqtt_client_host = dto.mqtt.client.host.as_ref();
+    mqtt_client_host
+        .map(|s| Arc::from(s.as_str()))
+        .unwrap_or(Arc::from("127.0.0.1"))
+}
+
```

Comment:
Would using `mqtt.client.host` as a default make sense here?

The problem is that there is a slight distinction between how `mqtt.client.host` and `http.client.host` is used in practice:
- `mqtt.client.host` should always be set on a child device because it's required to connect to the MQTT broker. There is no other channel where `mqtt.client.host` is passed to the child device by other thin-edge components, because to even begin to communicate with other thin-edge components this setting needs to be set. Setting this on a main device does not have any impact on child device behaviour.
- `http.client.host` setting is read by other thin-edge components, e.g. `c8y-configuration-plugin` and its value is put into MQTT messages directed towards the child device. This setting doesn't have to be set on a child device, but it must be set on the main device, so that the MQTT messages directed to the child device contain the correct address of the File Transfer Service.

This can potentially be very confusing to users, so perhaps some other name ought to be used.

Reply:
We can ignore the usage of `c8y-configuration-plugin` as this component has been deprecated.

Reply:
I wouldn't bother linking the `mqtt.client.host` and the `mqtt.bind.address` settings (when setting a default value). Both should use a default value of `127.0.0.1`.

Reply:
fixed d47bfa70eb596b901d90598432c806d058aa112a